flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -601,6 +636,11 @@ void markFinished() {
 							}
 						}
 
+						synchronized (accumulatorLock) {
```

Comment:
This lock here seems redundant. No place is looking for those two to be in sync.